edgelet/iotedge/src/logs.rs

```diff
@@ -28,15 +29,28 @@ impl<M> Logs<M> {
 
 impl<M> Command for Logs<M>
 where
-    M: 'static + ModuleRuntime + Clone,
+    M: 'static + ModuleRuntime + Clone + Send,
 {
     type Future = Box<dyn Future<Item = (), Error = Error> + Send>;
 
     fn execute(self) -> Self::Future {
         let id = self.id.clone();
-        let result = pull_logs(&self.runtime, &id, &self.options, stdout())
+        let result = self
```

Comment:
This is wrong for the same reason that `list_with_details` was created to solve - the module can be deleted between the `list` and the `pull_logs`.

Look at the error from `pull_logs` and use that to determine if the reason is that the module doesn't exist.

Reply:
Yeah... I considered that, but the error message seems a bit too generic to accurately reflect whether or not the module exists. Namely, when I debug print the error...

```
err = Error {
    inner: Client(Api(ApiError { code: 500, content: None }))
    
    Could not get logs for module tempSensor,
}
```

Reply:
I'm aware that's the issue you're trying to fix, yes. And I'm saying to fix the `pull_logs` function to return your `ErrorKind` instead of trying to divine it downstream (all the way from the mgmt server endpoint).

Also, you forgot to `git add` the `error.rs` file.

Reply:
>all it does is call `runtime.logs`

Hence why I said:

>>all the way from the mgmt server endpoint

Reply:
(resolving because my initial approach was abandoned)